### ISSUE TITLE
authentication not passed to /alert.json

### ISSUE THREAD

**Issue by roidelapluie** [prymitive/karma#24]
We authenticate to karma using apache

But now somehow with the rewrite to react, the /alerts.json is not authenticated like it was with unsee. In the console I see 401 errors on that endpoint.

**Comment by prymitive** [comment]
Can you share some apache configuration details? I can't reproduce it just by adding basic auth to the backend, once I login in the browser it passes basic auth down to the API correctly, so I must be missing something. Thanks

**Comment by prymitive** [comment]
Old (unsee) UI was using jquery `$.ajax()` calls while new UI (karma) uses [fetch()](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API), so there is definetely a difference.

**Comment by roidelapluie** [comment]
https://stackoverflow.com/questions/34558264/fetch-api-with-cookie


Fetch does not send cookies by default.

**Comment by prymitive** [comment]
Can you verify with browser web inspector if alerts.json requests are getting all headers you see on /?

**Comment by roidelapluie** [comment]
No, the cookie is not there. According to the web 'include' is mandatory for this.

**Comment by prymitive** [comment]
Added `{credentials: 'include'}`, docker image should be there in 15-30 minutes, please try and let me know if that doesn't fix it for you. Thanks!

**Comment by roidelapluie** [comment]
thanks. I do not use docker but will try to build a binary to test.

**Comment by prymitive** [comment]
It seems that passing `credentials: 'include'` means that CORS headers need to be updated:
```
The value of the 'Access-Control-Allow-Origin' header in the response must not be the wildcard '*' when the request's credentials mode is 'include'.
```

Right now there's `Access-Control-Allow-Origin: *`. Allowing `credentials: 'include'` might require setting origin in the config if we can't use `*`.

**Comment by prymitive** [comment]
I'm not sure if `credentials: 'include'` is the correct fix here, I think that default should work as it allows  credentials on the same origin, and I don't understand why wouldn't that be enough.
I need to look into it a little bit more

**Comment by prymitive** [comment]
Can you share some details of what's the setup you have? Are you using https://github.com/zmartzone/mod_auth_openidc?

**Comment by roidelapluie** [comment]
Yes with keycloak.

**Comment by roidelapluie** [comment]
still no lock in trying anything.

**Comment by prymitive** [comment]
FYI that change was included in 0.11 - https://github.com/prymitive/karma/releases/tag/v0.11
